### PR TITLE
Basic analog support

### PR DESCRIPTION
**Pull request by WillCMcC**
Add some basic analog support. I hooked into the existing button timeouts. 

Looking for some guidance on proper formatting rules here. Thanks for your work all!

### CODE REVIEW

File: src/battle-scene.ts [hunk 1]
```diff
@@ -93,7 +93,11 @@ export enum Button {
 	CYCLE_NATURE,
 	CYCLE_VARIANT,
 	SPEED_UP,
-	SLOW_DOWN
+	SLOW_DOWN,
+	LEFT_STICK_UP,
```
Comment (Flashfyre) [inline src/battle-scene.ts:97]: Do these need to be separate buttons for this to work? I feel like they should just trigger the directional buttons.

**Review by Flashfyre — CHANGES_REQUESTED** [review]

File: src/battle-scene.ts (hunk 1 above)
Comment (WillCMcC) [reply src/battle-scene.ts:97]: I should be able to refactor to just use the directional buttons

Comment (WillCMcC) [reply src/battle-scene.ts:97]: I think I can actually integrate this a little more deeply but we might lose some flexibility. I'll get it submitted today.

Comment (Flashfyre) [reply src/battle-scene.ts:97]: You might want to wait on that, someone is refactoring inputs entirely.

Comment (WillCMcC) [reply src/battle-scene.ts]: Ok, reworked this to use existing buttons and simplified it a bit.

Comment (WillCMcC) [reply src/battle-scene.ts]: hm, reusing the constants I seem to be getting phantom inputs, taking a look

Comment (Flashfyre) [reply src/battle-scene.ts]: The refactor is in, so this is ready to be updated

Comment (WillCMcC) [reply src/battle-scene.ts]: Roger, will take a look thanks. I'm on paternity so moving slowly

**Comment by bennybroseph** [comment]
Closing this due to inactivity. Feel free to make a new request if interest is reignited

**Pull request closed without merging**